bee-ledger/src/types/conflict.rs

```diff
@@ -0,0 +1,24 @@
+// Copyright 2020-2021 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+/// Represents the different reasons why a transaction can conflict with the ledger state.
+#[repr(u8)]
+#[derive(Debug, Copy, Clone, Eq, PartialEq)]
+pub enum ConflictReason {
+    /// The message has no conflict.
+    None = 0,
+    /// The referenced Utxo was already spent.
+    InputUtxoAlreadySpent = 1,
+    /// The referenced Utxo was already spent while confirming this milestone.
+    InputUtxoAlreadySpentInThisMilestone = 2,
+    /// The referenced Utxo cannot be found.
+    InputUtxoNotFound = 3,
+    /// The sum of the inputs and output values does not match.
+    InputOutputSumMismatch = 4,
+    /// The unlock block signature is invalid.
+    InvalidSignature = 5,
+    /// The dust allowance for the address is invalid.
+    InvalidDustAllowance = 6,
+    /// The semantic validation failed.
+    SemanticValidationFailed = 255,
```

Comment:
Does this need any more specifics? I imagine there's more than one reason that semantic validation can fail.

Reply:
The more specific semantic errors are actually all the other variants. This is kinda just a placeholder in case something unexpected happens. Also, this is part of the protocol so we can't really change it.

Reply:
Makes sense, I wondered if there was a reason for it 👍

Reply:
I can add a comment anyway